### PR TITLE
Fix value check in HeaderAccessUtils

### PR DESCRIPTION
**Pull request by snegrus-adb**
**Goals (and why)**: fixes https://github.com/palantir/atlasdb/issues/3440

**Implementation Description (bullets)**: use startsWith instead of contains for collection's items

**Testing (What was existing testing like?  What have you done to improve it?)**: added one test for that specific case

**Concerns (what feedback would you like?)**: would this make the code path slower?

**Where should we start reviewing?**: 

**Priority (whenever / two weeks / yesterday)**: not urgent

---
This change is [Reviewable](https://reviewable.io/reviews/palantir/atlasdb/3459)

### CODE REVIEW

**Comment by sandorw** [comment]
Can we use https://github.com/palantir/http-remoting/blob/develop/jaxrs-clients/src/main/java/feign/TextDelegateDecoder.java? Ideally we wouldn't need this logic specifically in AtlasDB at all.

File: docs/source/release_notes/release-notes.rst [hunk 1]
```diff
@@ -55,6 +55,11 @@ develop
            Previously, trying to refresh a larger number of locks could trigger the 50MB limit in payload size.
            (`Pull Request <https://github.com/palantir/atlasdb/pull/3450>`__)
 
+    *    - |fixed|
+         - Requests for which headers' values contain some metadata after the value are parsed correctly.
```
Comment (sandorw) [inline docs/source/release_notes/release-notes.rst]: Can you rephrase this as 'Now correctly parse ...' for clarity?

**Comment by snegrus-adb** [comment]
@sandorw , I've updated the class used with the one linked and updated the RN. I think more things could be replaced with the ones from http-remoting but that could be a separate PR. For `HeaderAccessUtils.java` the methods are slightly different. We have a method that checks if a given header has a given value but in the library it just checks if the header is present or returns its collection of values.

File: atlasdb-commons/src/main/java/com/palantir/common/remoting/HeaderAccessUtils.java [hunk 2]
```diff
@@ -48,7 +48,12 @@ public static boolean shortcircuitingCaseInsensitiveContainsEntry(
             String value) {
         for (Map.Entry<String, Collection<String>> entry : headers.entrySet()) {
             if (header.equalsIgnoreCase(entry.getKey())) {
-                return entry.getValue().contains(value);
+                for (String headerValue: entry.getValue()) {
```
Comment (sandorw) [inline atlasdb-commons/src/main/java/com/palantir/common/remoting/HeaderAccessUtils.java]: This method should be unused now aside from tests/benchmarks. Let's remove it.

File: atlasdb-commons/src/test/java/com/palantir/remoting/HeaderAccessUtilsTest.java [hunk 3]
```diff
@@ -97,6 +101,15 @@ public void caseInsensitiveGetShortcircuits() {
         assertEquals(VALUE_2, HeaderAccessUtils.shortcircuitingCaseInsensitiveGet(testMap, KEY_2.toUpperCase()));
     }
 
+    @Test
```
Comment (sandorw) [inline atlasdb-commons/src/test/java/com/palantir/remoting/HeaderAccessUtilsTest.java]: Shouldn't need this test, we can remove this method. We can probably drop the javax.ws.rs-api import as well, since I think this is the only place it's used.

File: atlasdb-commons/src/test/java/com/palantir/remoting/HeaderAccessUtilsTest.java [hunk 4]
```diff
@@ -97,6 +101,15 @@ public void caseInsensitiveGetShortcircuits() {
...
+    public void parseContentTypePlainText() {
+        Map<String, Collection<String>> testMap = Maps.newLinkedHashMap();
+        testMap.put("Content-Type", ImmutableList.of("text/plain;charset=iso-8859-1"));
```
Comment (sandorw) [inline atlasdb-commons/src/test/java/com/palantir/remoting/HeaderAccessUtilsTest.java]: nits, if we were going to keep this test:
* We imported HttpHeaders.CONTENT_TYPE, but still use the magic string "Content-Type" here
* Normal var naming convention in the next line is that all caps = static vars, and we should avoid naming a var the same thing as a static variable we're using

File: atlasdb-commons/src/main/java/com/palantir/common/remoting/HeaderAccessUtils.java (hunk 2 above)
Comment (sandorw) [reply atlasdb-commons/src/main/java/com/palantir/common/remoting/HeaderAccessUtils.java]: I think `shortcircuitingCaseInsensitiveContainsEntry` is completely unused now?

File: atlasdb-feign/src/main/java/com/palantir/atlasdb/http/TextDelegateDecoder.java [hunk 5]
```diff
@@ -1,71 +0,0 @@
...
-    private final Decoder stringDecoder;
-
-    public TextDelegateDecoder(Decoder delegate) {
-        this.delegate = delegate;
-        this.stringDecoder = new StringDecoder();
-    }
-
-    @Override
-    public Object decode(Response response, Type type) throws IOException, FeignException {
-        if (HeaderAccessUtils.shortcircuitingCaseInsensitiveContainsEntry(
```
Comment (sandorw) [inline atlasdb-feign/src/main/java/com/palantir/atlasdb/http/TextDelegateDecoder.java]: Please also remove `HeaderAccessUtils.shortcircuitingCaseInsensitiveContainsEntry`, since AFAICT it is now unused outside of tests and benchmarks. Please remove those tests and benchmarks as well.

File: atlasdb-feign/build.gradle [hunk 6]
```diff
@@ -28,6 +28,7 @@ dependencies {
   explicitShadow project(':lock-api')
   explicitShadow group: 'com.squareup.okhttp3', name: 'okhttp'
   explicitShadow group: 'com.palantir.remoting-api', name: 'service-config'
+  explicitShadow group: 'com.palantir.remoting3', name: 'jaxrs-clients'
```
Comment (sandorw) [inline atlasdb-feign/build.gradle]: I want to revisit if we need to shadow (and generally how much shadowing we need to do), but that's out of scope for this PR.
